src/arrangement/core.cljc

```diff
@@ -27,6 +27,18 @@
     (or priority (count predicates))))
 
 
+(defn type-name
+  "Get the type of the given object as a string. For Clojure, gets the name of
+  the class of the object. For ClojureScript, gets either the `name` attribute
+  or the protocol name if the `name` attribute doesn't exist."
+  [x]
+  #?(:clj (.getName (class x))
+     :cljs (let [t (type x)
+                 n (.-name t)]
+             (if (not-empty n)
```

Comment:
I prefer predicates in tests, especially since you can eliminate the `not` entirely by flipping the two results.